server/src/tokens.rs

```diff
@@ -311,4 +311,80 @@ mod tests {
 
         assert_eq!(actual, expected);
     }
+
```

Comment:
These aren't all that relevant but they were missing tests and nice to add while we're in this space